### ISSUE TITLE
rollback of a deleted release fails in helm v2.8.2

### ISSUE THREAD

**Issue by sami-alajrami** [helm/helm#3722]
**Problem**:
In version v2.8.2, when a release is deleted (not purged) can't be rolled back and gives the following error:

```
Error: "release-name" has no deployed releases
```
**Expected Behaviour:** 
To successfully roll back the release. This works fine on v2.8.1

**Environment**: 

```
$ helm version
Client: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}

$ kubectl version 
Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.6", GitCommit:"4bc5e7f9a6c25dc4c03d4d656f2cefd21540e28c", GitTreeState:"clean", BuildDate:"2017-09-14T06:55:55Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"8+", GitVersion:"v1.8.8-gke.0", GitCommit:"6e5b33a290a99c067003632e0fd6be0ead48b233", GitTreeState:"clean", BuildDate:"2018-02-16T18:26:58Z", GoVersion:"go1.8.3b4", Compiler:"gc", Platform:"linux/amd64"}
```

**Reproduce:**

```
helm install stable/jenkins --set Master.ServiceType=NodePort
...
helm list
NAME               	REVISION	UPDATED                 	STATUS  	CHART         	NAMESPACE
lolling-labradoodle	1       	Tue Mar 20 17:30:44 2018	DEPLOYED	jenkins-0.14.2	default  

helm delete lolling-labradoodle
release "lolling-labradoodle" deleted

helm list --deleted
NAME               	REVISION	UPDATED                 	STATUS 	CHART         	NAMESPACE
lolling-labradoodle	1       	Tue Mar 20 17:30:44 2018	DELETED	jenkins-0.14.2	default  

helm history lolling-labradoodle
REVISION	UPDATED                 	STATUS 	CHART         	DESCRIPTION      
1       	Tue Mar 20 17:30:44 2018	DELETED	jenkins-0.14.2	Deletion complete

helm rollback lolling-labradoodle 1
Error: "lolling-labradoodle" has no deployed releases
```

**Tiller Logs:**

``` 
[tiller] 2018/03/20 16:31:14 uninstall: Deleting lolling-labradoodle
[tiller] 2018/03/20 16:31:14 executing 1 pre-delete hooks for lolling-labradoodle
[tiller] 2018/03/20 16:31:14 hooks complete for pre-delete lolling-labradoodle
[storage] 2018/03/20 16:31:14 updating release "lolling-labradoodle.v1"
[kube] 2018/03/20 16:31:14 Starting delete for "lolling-labradoodle-jenkins-agent" Service
[kube] 2018/03/20 16:31:14 Starting delete for "lolling-labradoodle-jenkins" Service
[kube] 2018/03/20 16:31:14 Starting delete for "lolling-labradoodle-jenkins" Deployment
[kube] 2018/03/20 16:31:14 Using reaper for deleting "lolling-labradoodle-jenkins"
[kube] 2018/03/20 16:31:17 Starting delete for "lolling-labradoodle-jenkins" PersistentVolumeClaim
[kube] 2018/03/20 16:31:17 Starting delete for "lolling-labradoodle-jenkins" ConfigMap
[kube] 2018/03/20 16:31:17 Starting delete for "lolling-labradoodle-jenkins-tests" ConfigMap
[kube] 2018/03/20 16:31:17 Starting delete for "lolling-labradoodle-jenkins" Secret
[tiller] 2018/03/20 16:31:17 executing 1 post-delete hooks for lolling-labradoodle
[tiller] 2018/03/20 16:31:17 hooks complete for post-delete lolling-labradoodle
[storage] 2018/03/20 16:31:17 updating release "lolling-labradoodle.v1"
[storage] 2018/03/20 16:31:26 listing all releases with filter
[tiller] 2018/03/20 16:31:32 getting history for release lolling-labradoodle
[storage] 2018/03/20 16:31:32 getting release history for "lolling-labradoodle"
[tiller] 2018/03/20 16:31:41 preparing rollback of lolling-labradoodle
[storage] 2018/03/20 16:31:41 getting last revision of "lolling-labradoodle"
[storage] 2018/03/20 16:31:41 getting release history for "lolling-labradoodle"
[tiller] 2018/03/20 16:31:41 rolling back lolling-labradoodle (current: v1, target: v1)
[storage] 2018/03/20 16:31:41 getting release "lolling-labradoodle.v1"
[tiller] 2018/03/20 16:31:41 creating rolled back release for lolling-labradoodle
[storage] 2018/03/20 16:31:41 creating release "lolling-labradoodle.v2"
[tiller] 2018/03/20 16:31:41 performing rollback of lolling-labradoodle
[tiller] 2018/03/20 16:31:41 executing 1 pre-rollback hooks for lolling-labradoodle
[tiller] 2018/03/20 16:31:41 hooks complete for pre-rollback lolling-labradoodle
[kube] 2018/03/20 16:31:41 building resources from updated manifest
[kube] 2018/03/20 16:31:41 checking 7 resources for changes
[kube] 2018/03/20 16:31:41 Created a new Secret called "lolling-labradoodle-jenkins"
[kube] 2018/03/20 16:31:41 Created a new ConfigMap called "lolling-labradoodle-jenkins"
[kube] 2018/03/20 16:31:41 Created a new ConfigMap called "lolling-labradoodle-jenkins-tests"
[kube] 2018/03/20 16:31:41 Created a new PersistentVolumeClaim called "lolling-labradoodle-jenkins"
[kube] 2018/03/20 16:31:42 Created a new Service called "lolling-labradoodle-jenkins-agent"
[kube] 2018/03/20 16:31:42 Created a new Service called "lolling-labradoodle-jenkins"
[kube] 2018/03/20 16:31:42 Created a new Deployment called "lolling-labradoodle-jenkins"
[tiller] 2018/03/20 16:31:42 executing 1 post-rollback hooks for lolling-labradoodle
[tiller] 2018/03/20 16:31:42 hooks complete for post-rollback lolling-labradoodle
[storage] 2018/03/20 16:31:42 getting deployed releases from "lolling-labradoodle" history
```

**Comment by jascott1** [comment]
@bacongobbler is this behavior expected for 2.8.2?

**Comment by unguiculus** [comment]
I can confirm this issue. Rollback never works in 2.8.2. This is not expected behavior.

**Comment by unguiculus** [comment]
Just tested again with Helm 2.8.2 and Helm 2.9.0-rc3 taking back my above comment. It does work as expected. I think I had mixed up arguments.

**Comment by ErkiDerLoony** [comment]
I also see this issue with Helm 2.8.2. Helm creates a new release number which for ever remains in the state PENDING_ROLLBACK. The pods of the rolled back release are running, however.

**Comment by jascott1** [comment]
@ErkiDerLoony That seems like a different issue. Can you explain more?

**Comment by atmouse-** [comment]
Can not rollback to last successful revsion
```
root@hapi-6599bd8697-9h4ml:/app/data# /usr/bin/helm history --kube-context stable stable-friend
REVISION	UPDATED                 	STATUS          	CHART       	DESCRIPTION                                                 
1       	Sat Jul  7 09:12:18 2018	SUPERSEDED      	friend-0.1.0	Install complete                                            
2       	Mon Jul  9 03:26:19 2018	SUPERSEDED      	friend-0.1.0	Upgrade complete                                            
3       	Mon Jul  9 04:44:22 2018	FAILED          	friend-0.1.0	Upgrade "stable-friend" failed: Ingress.extensions "stabl...
4       	Mon Jul  9 04:57:00 2018	FAILED          	friend-0.1.0	Upgrade "stable-friend" failed: Ingress.extensions "stabl...
5       	Mon Jul  9 05:10:22 2018	SUPERSEDED      	friend-0.1.0	Upgrade complete                                            
6       	Mon Jul  9 05:10:49 2018	FAILED          	friend-0.1.0	Rollback "stable-friend" failed: Ingress.extensions "stab...
7       	Mon Jul  9 05:11:01 2018	PENDING_ROLLBACK	friend-0.1.0	Rollback to 2                                               
8       	Mon Jul  9 05:15:08 2018	PENDING_ROLLBACK	friend-0.1.0	Rollback to 2                                               
9       	Mon Jul  9 05:22:27 2018	PENDING_ROLLBACK	friend-0.1.0	Rollback to 1                                               
10      	Mon Jul  9 05:22:41 2018	PENDING_ROLLBACK	friend-0.1.0	Rollback to 1                                               
11      	Mon Jul  9 05:32:51 2018	PENDING_ROLLBACK	friend-0.1.0	Rollback to 1                                               
12      	Mon Jul  9 05:36:12 2018	PENDING_ROLLBACK	friend-0.1.0	Rollback to 1                                               
root@hapi-6599bd8697-9h4ml:/app/data# /usr/bin/helm version --kube-context stable 
Client: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
root@hapi-6599bd8697-9h4ml:/app/data# kubectl version --context stable
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.3", GitCommit:"2bba0127d85d5a46ab4b778548be28623b32d0b0", GitTreeState:"clean", BuildDate:"2018-05-21T09:17:39Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.3", GitCommit:"2bba0127d85d5a46ab4b778548be28623b32d0b0", GitTreeState:"clean", BuildDate:"2018-05-21T09:05:37Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
root@hapi-6599bd8697-9h4ml:/app/data# /usr/bin/helm rollback --kube-context stable stable-friend 1 --force
Error: "stable-friend" has no deployed releases
```

**Comment by withnale** [comment]
Any news on when rolling back deleted releases will be resolved? We've recently started trying to use this functionality again (which we used previously) and somewhat surprised this no longer works.

**Comment by floriantraber** [comment]
We just experienced an similar error. The rollback could not succeed because of a configuration error. Resulting in the same state as described above. Not a bug, just configuration error (duh don't use latest as a image-tag).

**Comment by cridam** [comment]
Any update on this? As already expressed in other issues, the result is the same with version 2.9.1 : rollback of a deleted release fails to PENDING_ROLLBACK.

**Comment by fsigalov** [comment]
I see the same issue in 2.10.0 also.

**Comment by farmerbean** [comment]
Same for 2.11.0

**Comment by bacongobbler** [comment]
closed via #4820
